### ISSUE TITLE
BitBucket dashboard widget - Password encryption error

### ISSUE THREAD

**Issue by asub927** [capitalone/Hygieia#1090]
Configured BitBucket collector locally and while configuring the widget in the dashboard with the repo and authentication details, the encryption password functionality is failing through the widget.  Attached is the screenshot for reference. 

![bitbucket](https://cloud.githubusercontent.com/assets/24964408/21944864/fae9e13e-d99d-11e6-8844-1ff3b1beaf12.jpg)

**Comment by satishc1** [comment]
@asub927 
Functionality merged in https://github.com/capitalone/Hygieia/pull/835
Refer to the "Encryption for private repos" section in the UI module readme.

**Comment by rdsteele** [comment]
I did follow the "Encryption for private repos" section in the UI module readme, as well as read many other issues that were related. I am also getting the error stated above.

Here are the steps I took:
C:\Hygieia-master\core\target>java -jar core-2.0.5-SNAPSHOT.jar com.capitalone.dashboard.util.Encryption
Your secret key is:
NH********************************TC
Sample encrypted string with the above key for 'thisIsMyPassword' is:
Fa+n*****************bpv

Property added to api.properties
key=NH********************************TC

Property added to Bitbucket collector file (git.properties)
git.key=NH********************************TC

So, what am I missing? Thanks

**Comment by satishc1** [comment]
@rdsteele Can you paste the command you used to startup the api along with the resulting log? If you can do the same with debug enabled that could help. I would like to see the error in the api log when you click the Save button on the repo widget.
@asub927 Let us know if it worked for you or if its still an issue.

**Comment by rdsteele** [comment]
Thanks for the quick response. Once I restarted my machine in order to gather the information you requested, the SCM widget accepted the password successfully without having to modify anything. Although I had restarted the processes several times today, it seems restarting my machine did the trick. I do see the credentials with an encrypted password in the database as well. Unfortunately, I'm getting '401 Unauthorized' from the Bitbucket collector when it attempts to retrieve commits from the repository. I'll do more research to see if I can identify the issue. Thanks again.

**Comment by rdsteele** [comment]
I was able to get the Bitbucket collector to work finally, but I did find an issue. The password I was using had a special character (#) which caused a problem with the encrypt/decrypt function. Once I switched to an alphanumeric password, the decryption worked and I was able to obtain the commit information. I'll open a new issue once I gather more information. Thanks again.

**Comment by asub927** [comment]
@rdsteele @satishc1 - I followed the steps for generating the secret key and updated the same in 
api properties file as well as bitbucket properties file but still getting the same error. 

I also changed the password to only contain alphanumeric but still getting the same encryption error. Could you please provide the detailed steps to resolve this issue?

Thanks for your help in advance.

**Comment by rdsteele** [comment]
So, my understanding is the password encryption error in the UI SCM widget configuration was solved by generating the secret key and adding that to the API properties file as 'key=secretkeyvalue'. This should allow the API to encrypt the password and store the credentials in the database. I also restarted my machine; stopping and starting the API process didn't seem to work for some reason, not sure why.

1. java -jar core-2.0.5-SNAPSHOT.jar com.capitalone.dashboard.util.Encryption
    Your secret key is:
    NH***************************TC

2. Property added to api.properties (or whatever your file is named)
    key=NH********************************TC

Adding the secret key to the Bitbucket properties allows the collector to read the encrypted password from the database, decrypt it and call Bitbucket. If this fails, you'll probably be getting a 401 Unathorized error, which is how I discovered the issue with the '#' special character.

3. Property added to Bitbucket git.properties  (or whatever your file is named)
    git.key=NH********************************TC
